operator/pkg/model/translation/cec_translator.go

```diff
@@ -318,6 +324,16 @@ func (i *cecTranslator) getClusters(m *model.Model) []ciliumv2.XDSResource {
 
 				if isGRPCService(m, ns, name, port) {
 					mutators = append(mutators, WithProtocol(HTTPVersion2))
+				} else if i.useAppProtocol {
+					appProtocol := getAppProtocol(m, ns, name, port)
```

Comment:
@sayboras I rebased to fix the conflicts and also slightly refactored the code because ifs were becoming a mess.